File: app/src/main/java/com/yao/zhihudaily/util/ResUtil.java
package com.yao.zhihudaily.util;


import android.annotation.SuppressLint;
import android.content.Context;
import android.content.res.Resources;

/**
 * @author Yao
 * @date 2016/8/17 0012
 */
public class ResUtil {

    @SuppressLint("StaticFieldLeak")
    private static Context sContext;

    public static void init(Context ctx) {
        sContext = ctx;
    }

    public static Resources getResources() {
        return sContext.getResources();
    }

    public static String getString(int resId) {
        return sContext.getResources().getString(resId);
    }

    public static String getString(int format, int resId) {
        return String.format(getString(format), getString(resId));
    }

    public static String getString(int format, String... str) {
        return String.format(getString(format), (Object[]) str);
    }

    public static String getString(String format, String... str) {
        return String.format(format, (Object[]) str);
    }

    public static String getString(String format, int resId) {
        return String.format(format, getString(resId));
    }

    public static int getColor(int resId) {
        return sContext.getResources().getColor(resId);
    }
}

File: app/src/main/java/com/yao/zhihudaily/util/ResUtil.kt
package com.yao.zhihudaily.util


import android.annotation.SuppressLint
import android.content.Context

/**
 * @author Yao
 * @date 2016/8/17 0012
 */
object ResUtil {

    @SuppressLint("StaticFieldLeak")
    private lateinit var sContext: Context

    fun init(ctx: Context) {
        sContext = ctx
    }

    fun getString(resId: Int): String {
        return sContext.resources.getString(resId)
    }

    fun getString(format: Int, resId: Int): String {
        return String.format(getString(format), getString(resId))
    }

    fun getString(format: Int, vararg str: String): String {
        return String.format(getString(format), *str as Array<Any>)
    }

    fun getString(format: String, vararg str: String): String {
        return String.format(format, *str as Array<Any>)
    }

    fun getString(format: String, resId: Int): String {
        return String.format(format, getString(resId))
    }

    fun getColor(resId: Int): Int {
        return sContext.resources.getColor(resId)
    }
}
